sktime/clustering/_base.py

```diff
@@ -115,7 +115,7 @@ def fit_predict(self, X: TimeSeriesInstances, y=None) -> np.ndarray:
         self.fit(X)
         return self.predict(X)
 
-    def predict_proba(self, X):
+    def predict_proba(self, X, n_clusters=None):
```

Comment:
yes, agree. No extra method specific params to the type defining methods.

Reply:
So it can be set for certain clusterers. The issue is not all clusterers take a parameter n_clusters and instead work it out. Additionally some clusteres (i.e. hierarchical) don't really have the concept of n_clusters at all because it's relative to where you are 'looking' and as such there isn't a n_clusters (google hierarchical dendrogram if you don't understand what I mean by 'looking'). 

Because of this I've choosen not to include n_clusters in the base class because it isnt relevant to all clusterers. As this variable doesn't exist in base I can't access the variable and therefore I attempt to infer it (which isnt always accurate because you can have empty clusters with no values in it so I have no way of knowing they exist) and therefore this parameter serves as a way of explicitly specifying the n_clusters.

@fkiraly probably knows best on this then: Is it worth having n_clusters in base and it being redundent for 10% of the clusterers but relevent for 90%?

Reply:
@chrisholder, I think we misunderstand each other!
I don't think @aiwalter or myself suggest to have this as a base class interface point.

It should be passed in `__init__`, for individual clusterers that use it, instead of in `predict` or `predict_proba`.